torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1552,7 +1552,10 @@ Tensor var_backward(
     // To apease ASAN
     auto n = self.numel();
     if (n == correction) {
-      return INFINITY * grad;
```

Comment:
May I know the context to why `inf` were being returned previously ? I think it should be `NaN` (undefined) instead. I had a quick test with `keepdim=True` which returns `NaN`, instead of `inf`.

```python
import torch

x = torch.zeros([1, 1, 1], requires_grad=True) 
y = x.var(dim=-1, keepdim=True, unbiased=True)
print("var(x): ", y) # tensor([[[nan]]], grad_fn=<VarBackward0>)
y.backward()
print("d var(x)/dx: ", x.grad) # tensor([[[nan]]])
```

Reply:
I think the whole special case path is to appease ASAN, i.e. we want to avoid division by zero.

Yes, NaN should be good IF correction is always 1 (becuase`self - self.mean()` is zero, and multiplying by zero by inf is NaN).

Unless we want to assume that, we should just do `std::numeric_limits<double>::infinity() * grad * (self - self.mean())` for this path as well. This also gives you the broadcast back to the correct size of grad for free.

Reply:
> Hopefully ASAN doesn't complain about doing infinity * 0 though...

It should not.

Reply:
Thanks @soulitzer, good point ! I didn't realize `correction` was an optional arg to `torch.var` (the doc seem to have missed mentioning it) !

Should we return `NaN` when `self == self.mean( )`, and otherwise return `inf`?

Reply:
@min-jean-cho  Yeah creating a mask from `self == self.mean()` and then generate the output using `where` also sounds good to me.

Reply:
That's a good idea ! Here is a sample output:

```
x = torch.tensor([1., 2., 3., 4., 5.], requires_grad=True)
y = x.var(correction=5)
y.backward()
>>> tensor([inf, inf, nan, inf, inf])
```